src/module/service.cpp

```diff
@@ -0,0 +1,290 @@
+#include <grpcpp/server.h>
+
+#include <csignal>
+#include <iostream>
+
+#include "component/generic/v1/generic.grpc.pb.h"
+#include "google/protobuf/descriptor.h"
+#define BOOST_LOG_DYN_LINK 1
+#include <app/v1/robot.pb.h>
+#include <grpcpp/channel.h>
+#include <grpcpp/create_channel.h>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/security/credentials.h>
+#include <grpcpp/security/server_credentials.h>
+#include <grpcpp/server_builder.h>
+#include <grpcpp/server_context.h>
+#include <grpcpp/support/status.h>
+#include <module/v1/module.grpc.pb.h>
+#include <module/v1/module.pb.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+
+#include <boost/log/trivial.hpp>
+#include <boost/none.hpp>
+#include <common/utils.hpp>
+#include <components/component_base.hpp>
+#include <components/generic.hpp>
+#include <components/service_base.hpp>
+#include <config/resource.hpp>
+#include <memory>
+#include <module/handler_map.hpp>
+#include <module/service.hpp>
+#include <registry/registry.hpp>
+#include <resource/resource.hpp>
+#include <resource/resource_base.hpp>
+#include <robot/client.hpp>
+#include <robot/service.hpp>
+#include <rpc/server.hpp>
+#include <string>
+#include <subtype/subtype.hpp>
+
+namespace {
+Dependencies get_dependencies(ModuleService_* m,
+                              google::protobuf::RepeatedPtrField<std::string> proto) {
+    Dependencies deps;
+    for (auto& dep : proto) {
+        auto name = Name::from_string(dep);
+        std::shared_ptr<ResourceBase> resource = m->get_parent_resource(name);
+        deps.emplace(name, resource);
+    }
+    return deps;
+}
+}  // namespace
+   //
+std::shared_ptr<ResourceBase> ModuleService_::get_parent_resource(Name name) {
+    if (parent == nullptr) {
+        parent = RobotClient::at_address("unix://" + parent_addr, {0, boost::none});
+    }
+
+    return parent->resource_by_name(name.to_proto());
+}
+
+::grpc::Status ModuleService_::AddResource(::grpc::ServerContext* context,
+                                           const ::viam::module::v1::AddResourceRequest* request,
+                                           ::viam::module::v1::AddResourceResponse* response) {
+    viam::app::v1::ComponentConfig proto = request->config();
+    Component cfg = Component::from_proto(proto);
+    std::shared_ptr<Module> module = this->module;
+    module->lock.lock();
+
+    std::shared_ptr<ResourceBase> res;
+    if (cfg.api.is_component_type()) {
+        std::shared_ptr<ComponentRegistration> reg = Registry::lookup_component(cfg.api, cfg.model);
+        if (reg != nullptr) {
+            res = reg->create_rpc_client(module->name, module->channel);
```

Comment:
we most likely need to use a different method here

Reply:
per above comment, will update to explicitly take `config` as an argument.